Route user profile

NandiniGera opened the pull request:
When we click on the username who has made a comment, it routes to that person's profile now!
I've tried working it on many users as well
Completes issue #796

NandiniGera commented:
Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment.
I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?

narayan954 commented:
> Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?

perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.

narayan954 commented:
> Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?

reason it's showing your profile data is because it's not getting external data and defaulting to logged in user data.
The best way to solve this issue imo is to implement dynamic routes asap for profile, so we can just redirect to the dynamic route of that user, and every other detail should be taken care of independently by that page

narayan954 commented on src/components/Post/CommentDialogBox.jsx:
```diff
@@ -44,15 +45,18 @@ const CommentDialogBox = ({
+                <span className="comment-doer-name" onClick={() => {
+                  navigate("/dummygram/profile", {
+                    state: {
+                      name: userComment.content.username,
```
photo and other details can be passed in this state object

NandiniGera replied:
> photo and other details can be passed in this state object

I tried doing that, but it didn't work, thats why i removed that code in the updated commit :/

![image](https://github.com/narayan954/dummygram/assets/97438986/af25f50d-6b3e-40f2-bb59-4121b608b4a9)

narayan954 replied:
ohhhh, perhaps some other issue, or component getting refreshed on its own. But dont worry now, route has been made dynamic and you dont need to pass these states at all now!! Amazing, isn't it? :)

narayan954 reviewed (changes requested):
I've marked where the problem currently is

NandiniGera commented:
> > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> 
> reason it's showing your profile data is because it's not getting external data and defaulting to logged in user data. The best way to solve this issue imo is to implement dynamic routes asap for profile, so we can just redirect to the dynamic route of that user, and every other detail should be taken care of independently by that page.
> 
> Till then, we can pass the state as in other components to the profile page while naviagting! :D

oohhh okayy got it:)

NandiniGera commented:
> I've marked where the problem currently is



> > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> 
> perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.

I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/

narayan954 commented:
> > I've marked where the problem currently is
> 
> > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > 
> > 
> > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> 
> I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/

listen, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal .
So you can just redirect to the correct username, rest is handled on its own :D

NandiniGera commented:
> > > I've marked where the problem currently is
> > 
> > 
> > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > 
> > > 
> > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > 
> > 
> > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> 
> listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D

okayy thats great!

narayan954 commented:
> > > > I've marked where the problem currently is
> > > 
> > > 
> > > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > > 
> > > > 
> > > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > > 
> > > 
> > > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> > 
> > 
> > listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D
> 
> okayy thats great!

Hi @NandiniGera is there any blocker?

NandiniGera commented:
> > > > > I've marked where the problem currently is
> > > > 
> > > > 
> > > > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > > > 
> > > > > 
> > > > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > > > 
> > > > 
> > > > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> > > 
> > > 
> > > listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D
> > 
> > 
> > okayy thats great!
> 
> Hi @NandiniGera is there any blocker?

Actually I updated the code, but it isn't working :/
I've added the commit

narayan954 commented:
> > > > > > I've marked where the problem currently is
> > > > > 
> > > > > 
> > > > > > > Actually, there's one issue in this- when we click on the username, it navigates to that user's profile correctly but on that user's profile it shows my profile pic(the user who has clicked that username who made the comment) only and not the profile pic of the one who made the comment. I guess the avatar is not fetched right now acc. to the code, I made another issue yesterday on changing the account circle icon with the user's profile pic in the comments, can I correct this with that issue itself?
> > > > > > 
> > > > > > 
> > > > > > perhaps, it might be due to the fact we're not passing those details in state while navigating, if I'm correct. If we pass the profile pic, it should render, right? Until we refresh the profile page, ofcourse.
> > > > > 
> > > > > 
> > > > > I added the state objects again, after refreshing it shows that user's profile pic just for one sec and then back again it shows my profile pic :/
> > > > 
> > > > 
> > > > listen,no need to worry now, we just implemented the major feature of dynamic profile routes, credits to @Palakkgoyal . So you can just redirect to the correct username, rest is handled on its own :D
> > > 
> > > 
> > > okayy thats great!
> > 
> > 
> > Hi @NandiniGera is there any blocker?
> 
> Actually I updated the code, but it isn't working :/ I've added the commit

aight checking

narayan954 reviewed (approved):
looks good to me!
